src/connection/host_lookup.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * host_lookup.c
+ */
+
+#ident "$Id$"
+
+#include <unordered_map>
+
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+#include <errno.h>
+
+#if defined (WINDOWS)
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#include <netdb.h>
+#endif
+
+#include "system_parameter.h"
+#include "environment_variable.h"
+
+#define HOSTNAME_BUF_SIZE              (256)
+#define MAX_HOSTS_LINE_NUM       (256)
+#define IPADDR_LEN              (17)
+
+#define IPADDR_STORE 0x0
+#define HOSTNAME_STORE 0x1
+
+typedef enum
+{
+  HOSTNAME_TO_IPADDR = 0,
+  IPADDR_TO_HOSTNAME = 1,
+} LOOKUP_TYPE;
+
+struct cub_hostent
+{
+  char ipaddr[IPADDR_LEN];
+  char hostname[HOSTNAME_BUF_SIZE + 1];
+};
+
+typedef struct cub_hostent CUB_HOSTENT;
+static CUB_HOSTENT hostent_List[MAX_HOSTS_LINE_NUM];	/* "etc/hosts" file maximum line */
+//should I need to allocate all 256 element??
+static const char user_defined_hostfile_Name[] = "hosts.conf";
+
+static int host_conf_element_Count = 0;
+static int host_conf_Use = -1;	/* -1 is unknown, 0 is not using user hosts, 1 is using user hosts */
+
+// *INDENT-OFF*
+static std::unordered_map <std::string, std::string> user_host_Map;
+// *INDENT-ON*
+
+static int host_conf_load ();
+static struct hostent *host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_case);
+
+static struct hostent *
+host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_case)
+{
+  static struct hostent hp;
+  int i, find_index = -1;
+
+  char addr_transform_buf[IPADDR_LEN];
+  char addr_transform_buf2[IPADDR_LEN];
+  struct sockaddr_in *addr_trans = NULL;
+
+  if (host_conf_element_Count == 0)
+    {
+      host_conf_element_Count = host_conf_load ();
+      if (host_conf_element_Count <= 0)
+	{
+	  return NULL;
+	}
+    }
+
+  addr_trans = (struct sockaddr_in *) saddr;
+
+  assert (((hostname != NULL) && (saddr == NULL)) || ((hostname == NULL) && (saddr != NULL)));
+
+  if (lookup_case == IPADDR_TO_HOSTNAME)
+    {
+      if (inet_ntop (AF_INET, &addr_trans->sin_addr, addr_transform_buf, sizeof (addr_transform_buf) + 1) == NULL)
```

Comment:
The length of addr_transform_buf is 17, and the length of sizeof (addr_transform_buf) + 1) is 18. The size does not match.

Reply:
Thank you. This is my mistake. 
This is purpose for the space appended characters like '\0'. But it is already considered by size 17.